src/perfoutputwidget.cpp

```diff
@@ -0,0 +1,198 @@
+bool PerfOutputWidgetKonsole::eventFilter(QObject* watched, QEvent* event)
+{
+    Q_UNUSED(watched);
+    if (event->type() == QEvent::KeyPress) {
+        auto keyEvent = static_cast<QKeyEvent*>(event);
+
+        // ignore ctrl + c
+        if (keyEvent->key() == Qt::Key_C && keyEvent->modifiers() == Qt::ControlModifier) {
+            return true;
+        }
+
+        if (m_inputEnabled) {
+            m_inputBuffer.append(keyEvent->text().toUtf8());
+
+            if (keyEvent->key() == Qt::Key_Enter) {
+                emit sendInput(m_inputBuffer);
+                m_inputBuffer.clear();
+                return false;
+            }
+        } else {
+            // eat all key events if input is disabled
+            return true;
+        }
+    } else if (event->type() == QEvent::MouseButtonPress) {
+        auto mouseEvent = static_cast<QMouseEvent*>(event);
+
+        // prevent the user from opening the context menu
+        if (mouseEvent->button() == Qt::MouseButton::RightButton) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void PerfOutputWidgetKonsole::addOutput(const QString& output)
+{
+    m_konsoleFile->write(output.toUtf8());
+    m_konsoleFile->flush();
+}
+
+void PerfOutputWidgetKonsole::clear()
+{
+    if (m_konsolePart) {
+        m_konsolePart->deleteLater();
+    }
+
+    m_konsolePart = createPart();
+    Q_ASSERT(m_konsolePart);
+
+    if (m_konsoleFile) {
+        m_konsoleFile->deleteLater();
+    }
+    m_konsoleFile = new QTemporaryFile(this);
+    m_konsoleFile->open();
+
+    const auto tail = findTail();
+    Q_ASSERT(!tail.isEmpty());
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(m_konsolePart);
+    if (!terminalInterface) {
+        qWarning("konsole kpart doesn't implement terminal interface");
+        delete m_konsolePart;
+        m_konsolePart = nullptr;
+        return;
+    }
+
+    terminalInterface->startProgram(tail, {tail, QStringLiteral("-f"), m_konsoleFile->fileName()});
+
+    m_konsolePart->widget()->focusWidget()->installEventFilter(this);
+
+    layout()->addWidget(m_konsolePart->widget());
+}
+
+void PerfOutputWidgetKonsole::enableInput(bool enable)
+{
+    if (enable != m_inputEnabled) {
+        m_inputEnabled = enable;
+        m_inputBuffer.clear();
+    }
+}
+
+void PerfOutputWidgetKonsole::setInputVisible(bool visible)
+{
+    Q_UNUSED(visible);
+}
+
+QString PerfOutputWidgetKonsole::findTail()
+{
+    const auto tail = QStandardPaths::findExecutable(QStringLiteral("tail"));
+
+    return tail;
+}
+
+KParts::ReadOnlyPart* PerfOutputWidgetKonsole::createPart()
+{
+    KService::Ptr service = KService::serviceByDesktopName(QStringLiteral("konsolepart"));
+
+    if (!service) {
+        return nullptr;
+    }
+    return service->createInstance<KParts::ReadOnlyPart>();
+}
+
+PerfOutputWidgetText::PerfOutputWidgetText(QWidget* parent)
+    : PerfOutputWidget(parent)
+{
+    auto layout = new QVBoxLayout(this);
+    layout->setMargin(0);
```

Comment:
? shouldn't this be `setContentMargins(0, 0, 0, 0)` - where did you find that function? 

https://doc.qt.io/qt-5/qvboxlayout-members.html

Reply:
autocomplete. I remembered that you had to set one margin to 0 but not witch one. So I choose the first one